### PR TITLE
[master] Fix XML parser truncating large texts 

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

Fixes https://github.com/wso2-enterprise/internal-support-ballerina/issues/416

## Approach
Improved the XML parser logic to return only when all the chars are combined. 
reference : https://stackoverflow.com/questions/29498587/xmlstreamreader-doesnt-read-complete-tag

## Samples
``` ballerina
import ballerina/io;
import ballerina/xmldata;
import ballerina/http;

type Reservation record {
    string reservationID;
    string confirmationID;   
};

listener http:Listener httpListener = new (8080);

service / on httpListener {
    resource function post hello(@http:Payload xml data) returns Reservation|error  {
        
        Reservation res = check  xmldata:fromXml(data);
        io:println(res);
        return res;
    }
}
```
The binding failed for the following request
```
POST http://localhost:8080/hello HTTP/1.1
Content-Type: application/xml

<Reservation>
<reservationID>1123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789aaaaaa</reservationID>
    <confirmationID>RPFABE</confirmationID>
</Reservation>
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/runtime/api/values/modules/xml_values/xml_values.bal [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```
Comment (gabilang) [inline tests/jballerina-unit-test/src/test/resources/test-src/runtime/api/values/modules/xml_values/xml_values.bal:1]:
```suggestion
// Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
The "All Rights Reserved" part needs to be removed as per the new license header for opensource codes.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlTreeBuilder.java [hunk 2]
```diff
@@ -121,8 +128,11 @@ public BXml parse() {
                         readComment(xmlStreamReader);
                         break;
                     case CDATA:
+                        readCData(xmlStreamReader);
```
Comment (hasithaa) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlTreeBuilder.java:131]: @prakanth97 Please note that this fix, We need to fix this in xmldata module too.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/runtime/api/tests/Values.java [hunk 3]
```diff
@@ -504,4 +506,27 @@ public static BError getErrorValue(BString errorTypeName) {
         Assert.assertEquals(bError.getType().getName(), errorTypeName.getValue());
         return bError;
     }
+
```
Comment (Nadeeshan96) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/runtime/api/tests/Values.java:509]:
```suggestion
```

**Review by gabilang — APPROVED** [review]
LGTM

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlTreeBuilder.java [hunk 4]
```diff
@@ -104,9 +104,16 @@ private void handleXMLStreamException(Exception e) {
...
+        int next;
         try {
             while (xmlStreamReader.hasNext()) {
-                int next = xmlStreamReader.next();
+                if (readNext) {
+                    readNext = false;
+                    next = xmlStreamReader.getEventType();
+                } else {
+                    next = xmlStreamReader.next();
+                }
```
Comment (prakanth97) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlTreeBuilder.java:116]: Shall we move this into a separate method

**Pull request merged**